### ISSUE TITLE
itd cleanup for CICE ICs

### ISSUE THREAD

**Issue by DeniseWorthen** [ufs-community/ufs-s2s-model#216]
**Description**
The current failure of CICE6 in the P5 runs may be related to the correct binning of thickness categories. 

The ICs created for CICE5/CICE6 are from non-CICE sources. These sources use a different definition of the ice thickness category boundaries. During the creating of the ICs, we map the categories 1:1 even though they are different than the categories we are using for CICE5/CICE6. 

EMC-CICE5 had an additional call added to the restart_driver routine to optionally call the cleanup_itd (cleanup ice-thickness-distribution) routine if the runtype was 'initial' and the runid was 'cpcice'. This was added during the debugging of the salt-flux fix and may or may not have been partially responsible for the stability of CICE5 in the benchmark runs.

**Solution**

There are two options:

1) Implement the cleanup-itd in CICE6 in the same manner as EMC-CICE5. This requires source code changes to CICE and may not be accepted back into the consortium code base. See related CICE issue [#20](https://github.com/NOAA-EMC/CICE/issues/20).

2) Implement a comparable re-binning of the thickness categories during the IC creation, either in the NCL script directly or as an additional using fortran code to mimic the cleanup_itd routine itself.

**Comment by DeniseWorthen** [comment]
**1**. I have gone back and examined when and why ITD was implemented in EMC-CICE5. At the time of BM3, there were two issues causing problems. 

-  Model failures in MOM6 with "excessive salt". This was the result of the salt flux bug. CICE exports Fsalt/Ai, expecting that the coupler re-applies the Ai prior to exporting to the ocean. The NEMS mediator was not doing this, so when Ai was very small very large salt fluxes were being sent to the ocean.  

- When using CPCice ICs, a few (<= 6) very large thicknesses would be generated in isolated locations along the coasts. 

It was because of the second issue that ITD clean-up was implemented. It is not clear in hindsight whether it would have been required if the salt-bug fix had been identified first.

**2.** The process of creating CICE ICs from either CFSR or CPC involves the following steps:

- mapping of 1/2 deg -> 1/4 deg 

- bucket-filling the 5 categories of CFSR/CPCice SIS/SIS2 into the 5 categories of CICE 1:1. The categories are not the same:
      SIS2 uses : 1e-10, 0.1, 0.3, 0.7, 1.1
     CICE5 uses (kcatbound=0): 0, 0.64, 1.39, 2.47, 4.57 

- for each category, linearly interpolating the 2 SIS/SIS2 layers to the 7 layers CICE

- creating an enthalpy value for all categories and layers. This is done by first using the linearly interpolated layer temperatures for each category and an idealized salinity profile to calculate qice for each layer and category.

- approximating or zero-ing out the remaining layer/category fields

If we believe that category mapping is important, then we should incorporate it at the start of the IC creation, not the end. If we do it at the start, we need only adjust a few fields (aicen, vicen etc). 

Performing ITD on the already-created 1/4 deg CICE5 restart files means having to adjust the entire suite of ncat,nlayer variables we've created. The possibility of introducing even more inconsistency in the restart file is substantial. For this reason, I don't believe applying ITD to the existing 1/4 deg CFSR/CPCice ICs is a reasonable option. 

To create new ICs with category remapping at the start of the IC creation would mean the entire set of CFSR and/or CPCice ICs would need to be retrieved and reprocessed. Also, I believe the initial conversion was done on Theia using a set of weights ("MOM0.5_to_MOM0.25_map.nc"). I don't know if these files exist on Hera. New mapping files may need to be generated (this is not difficult).

**3.** Using the existing ICs and adding "ITD" to the CICE code after the restart is read during model startup:

Because of the CICE code refactoring, introducing the calls to cleanup_itd it is not as straightforward as it was in EMC-CICE5. The amount of time to implement ITD in the code would be worth it if we were sure that ITD cleanup is the actual solution to the code failures. Because testing has shown that the code does not fail when run using the one-step coldstart, it is not clear to me that ITD is even the solution. 

Finally, while verbal reports have been made of other failures when using CICE6 in the coupled model, no github issue has been created in these cases. It is impossible to gauge how widespread or under what conditions these failures occur, so it is impossible to judge whether the time spent implementing ITD would be worth it.
